src/dodal/common/signal_utils.py

```diff
@@ -27,6 +31,40 @@ async def get_value(self) -> SignalDatatypeT:
         await self._update_value()
         return await super().get_value()
 
+    async def put(self, value: SignalDatatypeT | None, wait: bool) -> None:
+        if self.set_to_hardware_func:
+            write_value = self.initial_value if value is None else value
+            await self.set_to_hardware_func(write_value)
+        else:
+            return await super().put(value, wait)
+
+
+def create_rw_hardware_backed_soft_signal(
+    datatype: type[SignalDatatypeT],
+    get_from_hardware_func: Callable[[], Coroutine[Any, Any, SignalDatatypeT]],
+    set_to_hardware_func: SetHardwareType,
+    units: str | None = None,
+    precision: int | None = None,
+):
```

Comment:
Should: I think we should pass the initial_value as we are using it in put. same for the create_hardware_backed_soft_signal
```suggestion
def create_rw_hardware_backed_soft_signal(
    datatype: type[SignalDatatypeT],
    get_from_hardware_func: Callable[[], Coroutine[Any, Any, SignalDatatypeT]],
    set_to_hardware_func: Callable[[SignalDatatypeT], Coroutine[Any, Any, None]],
    initial_value: SignalDatatypeT | None = None,
    units: str | None = None,
    precision: int | None = None,
) -> SignalRW:
```

Reply:
Yh, I think it's an edge case anyway and `initial_value` is probably never going to be `None`